assets/chat/js/commands.test.js

```diff
@@ -0,0 +1,95 @@
+import ChatCommands, {
+  CHAT_COMMANDS,
+  NAME_MISSING_ERROR,
+  DESCRIPTION_MISSING_ERROR,
+} from './commands';
+
+test('Commands should be sorted alphabetically', () => {
+  expect(CHAT_COMMANDS).toStrictEqual(
+    [...CHAT_COMMANDS].sort((a, b) => a.name.localeCompare(b.name))
+  );
+});
+
+describe('Commands are correctly formatted', () => {
+  test.each(CHAT_COMMANDS)(
+    'Command name should be lowercase - %s',
+    (command) => {
+      expect(command.name).toBe(command.name.toLowerCase());
+    }
+  );
+  test.each(CHAT_COMMANDS)(
+    'Command description should end with a dot - %s',
+    (command) => {
+      expect(command.description.slice(-1)).toBe('.');
+    }
+  );
+});
+
+describe('Valid commands', () => {
+  test.each([
+    [
+      'Command with just a name and a description',
+      {
+        name: 'unignore',
+        description: 'Remove <nick> from your ignore list.',
+      },
+      ' /unignore - Remove <nick> from your ignore list. \r',
+    ],
+    [
+      'Command with one alias',
+      {
+        name: 'stalk',
+        description: 'Return a list of messages from <nick>.',
+        alias: ['s'],
+      },
+      ' /stalk, /s - Return a list of messages from <nick>. \r',
+    ],
+    [
+      'Command with multiple aliases',
+      {
+        name: 'message',
+        description: 'Send a whisper to <nick>.',
+        alias: ['msg', 'whisper', 'w', 'tell', 't', 'notify'],
+      },
+      ' /message, /msg, /whisper, /w, /tell, /t, /notify - Send a whisper to <nick>. \r',
+    ],
+    [
+      'Admin command',
+      {
+        name: 'mute',
+        description: 'Stop <nick> from sending messages.',
+        admin: true,
+      },
+      ' /mute - Stop <nick> from sending messages. \r',
+    ],
+  ])('%s', (_, command, expectedHelpString) => {
+    const chatCommandsGenerator = new ChatCommands();
+    expect(chatCommandsGenerator.formatHelpString(command)).toBe(
+      expectedHelpString
+    );
+  });
+});
+
+describe('Invalid commands', () => {
```

Comment:
Let's take a different approach here. To ensure the commands in `CHAT_COMMANDS` are always valid, add a test to simply loop through all of them and check that they have a name and a description. You took this approach with them being alphabetized and lowercase, so you might as well extend that. We'll eventually write a workflow that runs our unit tests for each PR so an invalid command never makes it into production.

With this approach, you don't need to check the validity of a command at runtime in `formatHelpString`, or create/export error messages for each kind of invalid command.